### PR TITLE
Rename `index_name` field to `data_set` in SearchRequest.

### PR DESCRIPTION
**Pull request by mansu**
Rename index_name field to data_set in SearchRequest so we can better clarify the intent of the field.

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ private EsSearchResponse doSearch(EsSearchRequest request) throws IOException {
     KaldbSearch.SearchRequest searchRequest = request.toKaldbSearchRequest();
     KaldbSearch.SearchResult searchResult = searcher.doSearch(searchRequest);
 
-    span.tag("requestIndexName", searchRequest.getIndexName());
+    span.tag("requestDataSet", searchRequest.getDataset());
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java:84]: nit: inconsistent spelling

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 2]
```diff
@@ -159,14 +159,14 @@ public static Collection<String> getSearchNodesToQuery(
       ServiceMetadataStore serviceMetadataStore,
       long queryStartTimeEpochMs,
       long queryEndTimeEpochMs,
-      String indexName) {
+      String dataSet) {
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:162]: nit: inconsistent spelling

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java [hunk 3]
```diff
@@ -347,15 +347,15 @@ private List<SearchResult<LogMessage>> distributedSearch(KaldbSearch.SearchReque
   }
 
   private List<KaldbServiceGrpc.KaldbServiceFutureStub> getSnapshotUrlsToSearch(
-      long startTimeEpochMs, long endTimeEpochMs, String indexName) {
+      long startTimeEpochMs, long endTimeEpochMs, String dataSet) {
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:350]: nit: inconsistent spelling

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/SearchQuery.java [hunk 4]
```diff
@@ -2,7 +2,8 @@
 
 /** A class that represents a search query internally to LogStore. */
 public class SearchQuery {
-  public final String indexName;
+  // TODO: Remove the dataSet field from this class since it is not a lucene level concept.
```
Comment (bryanlb) [inline kaldb/src/main/java/com/slack/kaldb/logstore/search/SearchQuery.java:5]: nit: inconsistent spelling

File: kaldb/src/main/proto/kaldb_search.proto [hunk 5]
```diff
@@ -5,8 +5,11 @@ package slack.proto.kaldb;
 option java_package = "com.slack.kaldb.proto.service";
 
 message SearchRequest {
-  string chunk_id = 1;
-  string index_name = 2;
+  // Data sets or chunk_ids to be searched
```
Comment (bryanlb) [inline kaldb/src/main/proto/kaldb_search.proto:8]: nit: inconsistent spelling

**Review by bryanlb — CHANGES_REQUESTED** [review]
Just some inconsistent spellings to cleanup, otherwise looks good.

File: kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java (hunk 1 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiService.java:84]: Fixed

File: kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java (hunk 2 above)
Comment (mansu) [reply kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java:162]: Fixed.

**Pull request merged**